Adding bottles_of_beer.rb to repo

anthonyfalzetti opened the pull request:
I ensured that this passed both Rubocop and Homework_tester. This is to complete homework assingment number 2.

[#88904054]

anthonyfalzetti committed:
Adding bottles_of_beer.rb to repo

I ensured that this passed both Rubocop and Homework_tester. This is to complete homework assingment number 2.

[#88904054]

jasonnoble commented on methods_and_recursion/bottles_of_beer.rb:
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env ruby
+
+def ending(number)
+  lyrics = <<lyrics
```
No need to create a variable here...

```
puts <<END_OF_LYRIC
....
END_OF_LYRIC
```

jasonnoble commented on methods_and_recursion/bottles_of_beer.rb:
```diff
@@ -0,0 +1,34 @@
+
+def single_bottle(number)
+  lyrics = <<lyrics
+#{number} bottles of beer on the wall, #{number} bottles of beer.
```
Lots of duplication here...  DRY up your code by making a method that takes number of bottles and returns that number with bottles or bottle?

jasonnoble commented:
Needs rebasing...

anthonyfalzetti committed:
Refactor bottles_of_beer to be more DRY...

In the process I had to create methods for just about everything possible.

[#88904054]

anthonyfalzetti committed:
Changed the program to include recursion.

 * I had to create a couple different methods to
keep from having too many lines in my methods 
 * Now the program will remove a number and determine
if it needs to run again 

[#88904054]
